### PR TITLE
Vijayasree/Terms and conditions link in staging redirecting to production page

### PR DESCRIPTION
**Pull request by vijayasree-deriv**
Changes:

-   ...

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by sean-binary — APPROVED** [review]

File: src/common/constants.ts [hunk 1]
```diff
@@ -25,10 +27,14 @@ export const domains = [
...
-    isBrowser() && window.location.hostname in domain_url_pair
-        ? domain_url_pair[window.location.hostname]
+const getDomainUrl = (): string => {
+    const var_staging_flag = isBrowser() ? localStorage.getItem('staging_demo') : null
+    const var_staging =
+        isBrowser() && null === var_staging_flag ? window.location.hostname : 'staging.deriv.com'
+    return isBrowser() && var_staging in domain_url_pair
+        ? domain_url_pair[var_staging]
         : deriv_com_url
+}
```
Comment (prince-deriv) [inline src/common/constants.ts:37]:
```suggestion
const getDomainUrl = (): string => {
    const staging_flag = localStorage.getItem('staging_demo') || null
    const current_url = staging_flag ? 'staging.deriv.com' : window.location.hostname 
    
    return isBrowser() && current_url in domain_url_pair
        ? domain_url_pair[current_url]
        : deriv_com_url
}
```


let's keep the variable names meaningful and simplier

File: src/common/constants.ts [hunk 2]
```diff
@@ -25,10 +27,14 @@ export const domains = [
     'deriv-com-git-fork',
 ] //deriv-com-git-fork for vercel server, localhost - for developer mode
 
-const getDomainUrl = (): string =>
-    isBrowser() && window.location.hostname in domain_url_pair
-        ? domain_url_pair[window.location.hostname]
+const getDomainUrl = (): string => {
+    const staging_flag = isBrowser() ? localStorage.getItem('staging_demo') : null
+    const current_url =
+        isBrowser() && null === staging_flag ? window.location.hostname : 'staging.deriv.com'
```
Comment (sean-binary) [inline src/common/constants.ts:33]:
```suggestion
    const current_url =
        isBrowser() && staging_flag === null  ? window.location.hostname : 'staging.deriv.com'
```

File: src/common/constants.ts [hunk 3]
```diff
@@ -25,10 +27,14 @@ export const domains = [
     'deriv-com-git-fork',
 ] //deriv-com-git-fork for vercel server, localhost - for developer mode
 
-const getDomainUrl = (): string =>
-    isBrowser() && window.location.hostname in domain_url_pair
-        ? domain_url_pair[window.location.hostname]
+const getDomainUrl = (): string => {
+    const staging_flag = isBrowser() ? localStorage.getItem('staging_demo') : null
```
Comment (sean-binary) [inline src/common/constants.ts:31]: [Clarification] This is solely for QA testing purposes?

Comment (vijayasree-deriv) [reply src/common/constants.ts:31]: yes

**Review by sean-binary — APPROVED** [review]
LGTM

**Review by sean-binary — APPROVED** [review]

**Pull request merged**